### PR TITLE
call onAuthStateChanged callback immediately

### PR DESCRIPTION
**Pull request by zeevl**
- Calls `onAuthStateChanged` observer immediately, with the `currentUser` (see https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseAuth.AuthStateListener)
- Adds tests for `onAuthStateChanged`
- Adds debug config for vs code. :)

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (dmurvihill) [inline .vscode/launch.json]: I don't think you meant to include this. Might want to add `.vscode/` to your [global excludes file](https://gist.github.com/subfuzion/db7f57fff2fb6998a16c).

File: test/unit/auth.js [hunk 2]
```diff
@@ -244,9 +244,8 @@ describe('Auth', function () {
       ref.onAuth(spy);
       expect(spy).to.have.been.calledWith(null);
     });
-
   });
-
+
```
Comment (dmurvihill) [inline test/unit/auth.js]: Oops!

**Comment by dmurvihill** [comment]
Fantastic tests though. Was definitely thrilled to see this one, especially since I remember tripping over this issue myself 👍

File: .vscode/launch.json (hunk 1 above)
Comment (zeevl) [reply .vscode/launch.json]: Up to you if you wanted it.. it's handy for anyone contributing to this that's using vs code.   Sounds like you don't, so added .vscode to `.gitignore` instead.

**Comment by zeevl** [comment]
Seems odd to work around it since it's consistent with the core firebase auth, but up to you. :)

**Pull request merged**